torch_xla/core/c10d_registration.py

```diff
@@ -0,0 +1,25 @@
+import torch
+import torch_xla.runtime as xr
+import torch_xla.core.xla_model as xm
+from torch.library import Library
+from typing import List, Optional, TypedDict
+
+c10d_lib = Library("_c10d_functional", "IMPL")
+
+
+# "broadcast(Tensor self, int src, str tag, int[] ranks, int group_size) -> Tensor",
+@torch.library.impl(c10d_lib, "broadcast", "XLA")
+def broadcast_xla(self: torch.Tensor,
+                  src: int,
+                  tag: str,
+                  ranks: Optional[List] = None,
+                  group_size: Optional[int] = None) -> torch.Tensor:
+  assert group_size == None, "currently does not support group_size"
+  # xm.collective_broadcast perform an inplace update, but
```

Comment:
Can you clone the input and do the inplace `xm.collective_broadcast` on that instead of copying the implementation?

Either way, we should do proper functional implementations of these ops in C++ along with `all_reduce` (cc @zpcore since we talked about this today)

Reply:
I really want to just get rid of the old `collective_broadcast`, it is an inplace op so it should be `collective_broadcast_`. Given that @zpcore is going to register theses op properly I think we should delete `xm.collective_broadcast` at some point so it is better not to depend this implementation on that one.